File: hw/Pérez/U3/HW28AbstractFactorySingleton/AbstractFactory/src/main/java/ec/edu/espe/abstractfactory/model/LinuxFactory.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.abstractfactory.model;

/**
 *
 * @author Carlos Perez,The Javas, DCCO-ESPE
 */
public class LinuxFactory extends GUIFactory{
    
    public Button createButton() {
        System.out.println("Creating Linux Button");
        return (new LinuxButton());
    }

    
    public Menu createMenu() {
        System.out.println("Creating Linux Menu");
        return (new LinuxMenu());
    }
    
}

File: hw/villagomez/u3/hw28SingletonAndAbstractFactory/AbstractFactoryJS/model/LinuxFactory.js
import { GUIFactory } from './GUIFactory.js';
import { LinuxButton } from './LinuxButton.js';
import { LinuxMenu } from './LinuxMenu.js';

export class LinuxFactory extends GUIFactory {
    createButton() {
        console.log("Creating Linux Button");
        return new LinuxButton();
    }

    createMenu() {
        console.log("Creating Linux Menu");
        return new LinuxMenu();
    }
}
